aws/resource_aws_pinpoint_app_test.go

```diff
@@ -2,27 +2,74 @@ package aws
 
 import (
 	"fmt"
-	"os"
+	"log"
 	"testing"
 
-	"github.com/aws/aws-sdk-go/service/pinpoint"
-
 	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/pinpoint"
 	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/terraform"
 )
 
-func TestAccAWSPinpointApp_basic(t *testing.T) {
-	oldDefaultRegion := os.Getenv("AWS_DEFAULT_REGION")
-	os.Setenv("AWS_DEFAULT_REGION", "us-east-1")
-	defer os.Setenv("AWS_DEFAULT_REGION", oldDefaultRegion)
+func init() {
+	resource.AddTestSweepers("aws_pinpoint_app", &resource.Sweeper{
+		Name: "aws_pinpoint_app",
+		F:    testSweepPinpointApps,
+	})
+}
+
+func testSweepPinpointApps(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).pinpointconn
+
+	input := &pinpoint.GetAppsInput{}
+
+	for {
+		output, err := conn.GetApps(input)
+		if err != nil {
+			if testSweepSkipSweepError(err) {
+				log.Printf("[WARN] Skipping Pinpoint app sweep for %s: %s", region, err)
+				return nil
+			}
+			return fmt.Errorf("Error retrieving Pinpoint apps: %s", err)
+		}
+
+		if len(output.ApplicationsResponse.Item) == 0 {
+			log.Print("[DEBUG] No Pinpoint apps to sweep")
+			return nil
+		}
 
+		for _, item := range output.ApplicationsResponse.Item {
+			name := aws.StringValue(item.Name)
+
+			log.Printf("[INFO] Deleting Pinpoint app %s", name)
+			_, err := conn.DeleteApp(&pinpoint.DeleteAppInput{
+				ApplicationId: item.Id,
+			})
+			if err != nil {
+				return fmt.Errorf("Error deleting Pinpoint app %s: %s", name, err)
+			}
+		}
+
+		if output.ApplicationsResponse.NextToken == nil {
+			break
+		}
+		input.Token = output.ApplicationsResponse.NextToken
+	}
+
+	return nil
+}
+
+func TestAccAWSPinpointApp_basic(t *testing.T) {
 	var application pinpoint.ApplicationResponse
-	resourceName := "aws_pinpoint_app.test_app"
+	resourceName := "aws_pinpoint_app.test"
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:      func() { testAccPreCheck(t) },
+		PreCheck:      func() { testAccPreCheck(t); testAccRegionHasServicePreCheck("pinpoint", t) },
```

Comment:
Need to explicitly pre-check that the test region has the Pinpoint service instead of just failing the test as the Pinpoint APIs hang in unsupported regions rather than return errors.